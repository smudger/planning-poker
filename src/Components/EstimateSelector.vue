<script setup>
import { RadioGroup, RadioGroupLabel, RadioGroupOption } from "@headlessui/vue";

defineProps({
  options: { type: Array, required: true },
});

defineEmits(["change"]);
</script>

<template>
  <RadioGroup
    class="fixed bottom-0 inset-x-0 flex items-center justify-center px-4 py-6"
    name="estimate"
    @update:model-value="$emit('change', $event)"
  >
    <RadioGroupLabel class="sr-only">Estimate</RadioGroupLabel>
    <div class="grid grid-cols-4 gap-3 sm:grid-cols-6 lg:grid-cols-12">
      <RadioGroupOption
        v-for="option in options"
        :key="option"
        v-slot="{ active, checked }"
        :value="option"
        as="template"
      >
        <div
          :class="[
            'cursor-pointer focus:outline-none',
            active ? 'ring-2 ring-indigo-600 ring-offset-2' : '',
            checked
              ? 'bg-indigo-600 text-white hover:bg-indigo-500'
              : 'ring-1 ring-inset ring-gray-300 bg-white text-gray-900 hover:bg-gray-50',
            'flex items-center justify-center rounded-md px-6 py-8 text-sm font-semibold uppercase sm:flex-1',
          ]"
        >
          <RadioGroupLabel as="span">{{ option }}</RadioGroupLabel>
        </div>
      </RadioGroupOption>
    </div>
  </RadioGroup>
</template>
