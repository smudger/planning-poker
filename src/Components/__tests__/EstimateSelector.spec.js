import { describe, expect, it } from "vitest";
import { shallowMount } from "@vue/test-utils";
import EstimateSelector from "@/Components/EstimateSelector.vue";

describe("EstimateSelector", () => {
  it("displays a radio button for each option", () => {
    const options = [1, 2, 3];

    const wrapper = shallowMount(EstimateSelector, {
      props: { options },
    });

    const radios = wrapper.findAll('input[type="radio"]');
    expect(radios.length).toBe(3);
    expect(radios.map((radio) => parseInt(radio.element.value))).toEqual(
      options,
    );
  });

  it("emits an event when the estimate is changed", async () => {
    const options = [1, 2, 3];

    const wrapper = shallowMount(EstimateSelector, {
      props: { options },
    });

    await wrapper.find('input[type="radio"][value="2"]').setValue();

    expect(wrapper.emitted()["change"][0][0]).toBe(2);

    await wrapper.find('input[type="radio"][value="3"]').setValue();

    expect(wrapper.emitted()["change"][1][0]).toBe(3);
  });
});
