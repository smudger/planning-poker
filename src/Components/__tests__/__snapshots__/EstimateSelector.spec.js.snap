// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`EstimateSelector > renders correctly 1`] = `
<div
  data-v-app=""
>
  
  <div
    class="fixed bottom-0 inset-x-0 flex items-center justify-center px-4 py-6"
    id="headlessui-radiogroup-17"
    role="radiogroup"
  >
    <label
      class="sr-only"
      id="headlessui-label-18"
    >
      Estimate
    </label>
    <div
      class="grid grid-cols-4 gap-3 sm:grid-cols-6 lg:grid-cols-12"
    >
      
      <div
        aria-checked="false"
        class="cursor-pointer focus:outline-none ring-1 ring-inset ring-gray-300 bg-white text-gray-900 hover:bg-gray-50 flex items-center justify-center rounded-md px-6 py-8 text-sm font-semibold uppercase sm:flex-1"
        id="headlessui-radiogroup-option-19"
        role="radio"
      >
        <span
          id="headlessui-label-20"
        >
          1
        </span>
      </div>
      <div
        aria-checked="false"
        class="cursor-pointer focus:outline-none ring-1 ring-inset ring-gray-300 bg-white text-gray-900 hover:bg-gray-50 flex items-center justify-center rounded-md px-6 py-8 text-sm font-semibold uppercase sm:flex-1"
        id="headlessui-radiogroup-option-21"
        role="radio"
      >
        <span
          id="headlessui-label-22"
        >
          2
        </span>
      </div>
      <div
        aria-checked="false"
        class="cursor-pointer focus:outline-none ring-1 ring-inset ring-gray-300 bg-white text-gray-900 hover:bg-gray-50 flex items-center justify-center rounded-md px-6 py-8 text-sm font-semibold uppercase sm:flex-1"
        id="headlessui-radiogroup-option-23"
        role="radio"
      >
        <span
          id="headlessui-label-24"
        >
          3
        </span>
      </div>
      
    </div>
  </div>
  
</div>
`;
